### PR TITLE
completed preorder CRUD

### PR DESCRIPTION
**Pull request by JorjaF**
## Implements/Fixes: title-here
Added CRUD for preorders

## This closes issue:
#28 #29 #30 #31 #32 

## Neccesary checkmarks:
- [ ] All Tests are Passing
- [x ] The code will run locally

## Type of change
- [ x] New feature
- [ ] Bug Fix

## Check the correct boxes
- [ x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:
- [x ] My code has no unused/commented out code
- [ x] I have reviewed my code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

TY!

### CODE REVIEW

File: market_next_door_api/market_next_door_api/serializers.py [hunk 1]
```diff
@@ -34,4 +34,23 @@ def to_representation(self, instance):
...
+    data = {
+      "id": representation['id'],
+      "type": "preorder",
+      "attributes": {
+        "customer": representation['customer'],
+        "item": representation['item'],
+        "ready": representation['ready']
+      }
+    }
+    return data
```
Comment (EVanGorkom) [inline market_next_door_api/market_next_door_api/serializers.py:56]: We'll want to delete this section for the serializer. We agreed earlier that we'll just give FE the simplest version of a response as we can generate (without extra formatting). So we just need lines 39 - 43 now.

**Review by EVanGorkom — CHANGES_REQUESTED** [review]
This looks great besides that extra method in the serializer now. One quick fix and we'll have all of our model's CRUD functions and endpoints ready!

File: market_next_door_api/market_next_door_api/views.py [hunk 2]
```diff
@@ -214,3 +214,55 @@ def vendor_details(request, id):
...
+def update_preorder(preorder, data):
+  preorder_data = PreorderSerializer(preorder, data=data)
+  if preorder_data.is_valid():
+    preorder_data.save()
+    return Response(preorder_data.data)
+  return Response(preorder_data.errors, status=status.HTTP_400_BAD_REQUEST)
+
+def delete_preorder(preorder):
+  preorder.delete()
+  return Response(status=status.HTTP_204_NO_CONTENT)
```
Comment (kbug819) [inline market_next_door_api/market_next_door_api/views.py:268]: Awesome job on this Jorja! Great job getting this functionality mapped and created!

File: market_next_door_api/market_next_door_api/serializers.py (hunk 1 above)
Comment (kbug819) [reply market_next_door_api/market_next_door_api/serializers.py:56]: Agreed, let's simplify the serialization.

**Review by kbug819 — APPROVED** [review]

**Comment by EVanGorkom** [comment]
Thank you for the hotfix!

**Pull request merged**